tools/dm/dm-upgrade.md

```diff
@@ -0,0 +1,144 @@
+    ```
+    
+    and modify
+    
+    ```ini
+    dm_master ansible_host=172.16.10.71 dm_master_port=12080 dm_master_status_port=12081	
+    ```
+    
+    to
+    
+    ```ini
+    dm_master ansible_host=172.16.10.71 dm_master_port=8261
+    ```
+    
+2. Use DM-Ansible to perform rolling update on DM, Prometheus and Grafana.
+
+## Upgrade to v1.0.0-133-g2f9fe82
+
+### Version information
+
+```bash
+Release Version: v1.0.0-133-g2f9fe82
+Git Commit Hash: 2f9fe827d668add6493b2a3da107e0a01b94c6d1
+Git Branch: dm-master
+UTC Build Time: 2018-12-19 04:58:46
+Go Version: go version go1.11.2 linux/amd64
+```
+
+### Main changes
+
+Before this version, `mysql-instances` in the task configuration file (`task.yaml`) contains the following information:
+
+- `config`: the upstream MySQL address, user name and password.
+- `instance-id`: identifies an upstream MySQL.
+
+Since this version, the above two kinds of information are removed and the following information is added:
+
+- `source_id`: exists in `inventory.ini`, used to identify an upstream MySQL instance or a master-slave replication group.
+- `source-id`: exists in `mysql-instances` of the task configuration file. Its value corresponds to `source_id` of `inventory.ini`.
+
+> **Note:** To guarantee that the checkpoint information of the existing task stored in the downstream database can still be used, the value of `source_id`or `source-id` should be consistent with that of `instance-id` of the corresponding DM-worker.
+
+### Upgrade operation example
+
+1. Modify the `inventory.ini` configuration information.
+
+    Set the corresponding `source_id` for all DM-worker instances.
+    
+    For example, modify
+
+    ```ini
+    dm-worker1 ansible_host=172.16.10.72 server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+
+    to
+
+    ```ini
+    dm-worker1 ansible_host=172.16.10.72 source_id="mysql-replica-01" server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+
+2. Use DM-Ansible to perform rolling update on DM.
+
+3. Modify the task configuration file (`task.yaml`).
+    
+    Remove the `config` and `instance-id` configuration items and add the `source-id` configuration item (corresponding to `source_id` in `inventory.ini`).
+    
+    For example, modify
+    
+    ```yaml
+    config:
+          host: "192.168.199.118"	
+          port: 4306	
+          user: "root"	
+          password: "1234"	
+    instance-id: "instance118-4306" # It is unique. It is used as the ID when storing the checkpoint, configuration and other information.
+    ```
+    
+    to
+    
+    ```yaml
+    source-id: "instance118-4306" # It should be consistent with the original `source_id` value when the checkpoint of the original task needs to be reused.
+    ```
+
+4. Use the modified task configuration to restart the task.
```

Comment:
I think it's clear enough to remove the beginning section because the configuration is modified in the last step.

```suggestion
4. Restart the task.
```